src/GraphQL/Execution/ExecutionContext.cs

```diff
@@ -110,5 +110,12 @@ protected virtual void ClearContext()
                 _trackedArrays.Clear();
             }
         }
+
+        /// <summary>
+        /// Allows for an execution strategy to reuse an instance of <see cref="ReadonlyResolveFieldContext"/>.
+        /// This field may be accessed by multiple threads at the same time, so
+        /// access is restricted to <see cref="System.Threading.Interlocked.Exchange{T}(ref T, T)"/>.
```

Comment:
```suggestion
        /// access is restricted to <see cref="System.Threading.Interlocked.CompareExchange{T}(ref T, T, T)"/>.
```

Reply:
Well, it's using both...

Reply:
And we can't use CompareExchange in the other place.

Reply:
```suggestion
        /// access is restricted to <see cref="System.Threading.Interlocked.Exchange{T}(ref T, T)"/>
        /// and <see cref="System.Threading.Interlocked.CompareExchange{T}(ref T, T, T)"/>.
```

Reply:
I had used Exchange in both places to simply the notes more than anything else.  But if you want CompareExchange where it returns a copy to the "pool", then it is necessary to use both Exchange and CompareExchange.

Reply:
I forgot about both methods. Do as you see fit.